src/main/java/org/mskcc/cmo/metadb/service/impl/CmoLabelGeneratorServiceImpl.java

```diff
@@ -159,7 +158,8 @@ public String generateCmoSampleLabel(String requestId, IgoSampleManifest sampleM
         }
 
         // get next incremement value for cmo sample counter
-        Integer nextSampleCounter = getNextSampleIncrement(existingSamples);
+        Integer nextSampleCounter = getNextSampleIncrement(sampleManifest.getIgoId(), existingSamples);
```

Comment:
first check for matches against the existingSamples by the igo/primary id here. if a match is found then extract the sample counter from the matching sample's cmo label and use that - otherwise call the `getNextSampleIncrement`

I feel like relying on the increment/decrement of the counter might be buggy down the line